wiki/Bot-Fontend.md

```diff
@@ -0,0 +1,116 @@
+# Bot Frontend
+
+The Bot Frontend is the primary user interface of MODiX. It is the user interface that is accessed by users from within the Discord Web or Desktop application.
+
+## Purpose
+
+1. Provide simple, conveniently-accessible, user interfaces for MODiX functionality
+2. Serve as the primary user interface for most day-to-day operations
+3. Allow for command and rendering logic to be tested, independently from other layers of the application
+
+## Methodology
+
+### Commands
+
+A command is a UI element that mimics a classical Command-Line Interface (CLI), consisting of an identifying prefix, a name, and any number of arguments, all provided through text. In the context of Discord.NET, commands are implemented through a listener, which checks every message received through the Discord Gateway connection for the presence of the identifying prefix (a `'!'` character, or other character set through config, or a mention of @MODiX itself). Messages that contain the identifying prefix are passed to Discord.NET's `ComandService` which parses the text to determine the name and arguments, and (if valid) invokes the corresponding Command method upon a Module class.
+
+Commands are implemented in Discord.NET by subclassing `ModuleBase`, implementing methods to be invoked when the commands are executed, and annotating the class, the method, and its parameters with various Discord.NET attributes. These attribute serve to identify the command for the Discord.NET reflection system that maps available commands in memory to executable code, and define command metadata that is used by the Command Help system.
+
+Generally, command methods should contain almost exclusively rendering code. Commands should generally invoke the [Business Logic](Business-Logic) layer to perform all the real work, by invoking service methods, or publishing request objects. Command results are also generally rendered into embeds.
+
+Commands should always provide some kind of feedback to the user, especially when it does not produce any result data. This can be accomplished by attaching a reaction to the command message itself, indicating success or failure. The MODiX command system already implements rudimentary error handling and response for exceptions thrown by a command, in the form of an attached reaction that can be clicked to reveal a more detailed error message. Commands that may be long-running should also provide immediate confirmation that the command was received, by attaching a reaction that indicates the command was received, and is processing.
```

Comment:
Plural/singular mismatch: "Commands should always ... especially when it does not produce any result data." -> "Commands should always ... especially when they do not produce any result data."

Repetitive phrasing: "Commands ... immediate confirmation that the command was received, by attaching a reaction that indicates the command was received, and is processing." -> "Commands ... immediate confirmation that the command was received and is processing by attaching a reaction."

```suggestion
Commands should always provide some kind of feedback to the user, especially when they do not produce any result data. This can be accomplished by attaching a reaction to the command message itself, indicating success or failure. The MODiX command system already implements rudimentary error handling and response for exceptions thrown by a command, in the form of an attached reaction that can be clicked to reveal a more detailed error message. Commands that may be long-running should also provide immediate confirmation that the command was received and is processing by attaching a reaction.
```